test/test_utils.js

```diff
@@ -74,6 +74,12 @@ describe("Testing utils functions", function() {
             assert.equal(utils.readable_msisdn("0821234567", "+27"), "0821234567");
             assert.equal(utils.readable_msisdn("+27821234567", "27"), "0821234567");
         });
+        // the next test case should not occur as the msisdn's passed in should
+        // either be normalized (start with '+') or as is (start with '0')
+        it("should return msisdn as is if contains no leading +", function() {
+            assert.equal(utils.readable_msisdn("27821234567", "+27"), "27821234567");
+            assert.equal(utils.readable_msisdn("27821234567", "27"), "27821234567");
```

Comment:
I don't know what you just said - my head is too full of other stuff. What I do know is that `readable_msisdn` should present a readable msisdn - i.e. one without country code in the front. In South Africa we always want to present a 10-digit number starting with `0`. Can't think about other countries now - I'll leave that up to you.